### PR TITLE
CC-278: further improvements to logical type support

### PR DESCRIPTION
**Pull request by shikhar**
- Propagate schema parameters through as they may be required for determining the correct datatype mapping (e.g. scale for Decimal). To do this cleanly refactored DbDialect from having multiple maps to an overridable method for going from (schema name, schema parameters, schema type) -> dialect-specific sql type.
- Allow for formatting of default values with an overridable method, as this is dialect-specific. Added implementation for PostgreSQL logical types which were added in #135.

### CODE REVIEW

**Comment by shikhar** [comment]
ping @confluentinc/connect @willyhoang for review

**Comment by shikhar** [comment]
linking #133

File: src/main/java/io/confluent/connect/jdbc/sink/dialect/DbDialect.java [hunk 1]
```diff
@@ -109,20 +107,20 @@ public void apply(StringBuilder builder, SinkRecordField f) {
...
+      formatColumnValue(builder, f.schemaName(), f.schemaParameters(), f.schemaType(), f.defaultValue());
+    } else if (f.isOptional()) {
       builder.append(" NULL");
     } else {
       builder.append(" NOT NULL");
     }
   }
 
-  static void formatColumnValue(StringBuilder builder, Schema.Type type, Object value) {
+  protected void formatColumnValue(StringBuilder builder, String schemaName, Map<String, String> schemParameters, Schema.Type type, Object value) {
```
Comment (willyhoang) [inline src/main/java/io/confluent/connect/jdbc/sink/dialect/DbDialect.java]: I think you meant to name the parameter "schemaParameters".

File: src/main/java/io/confluent/connect/jdbc/sink/dialect/SqlServerDialect.java [hunk 2]
```diff
@@ -31,7 +31,7 @@
 public class SqlServerDialect extends DbDialect {
 
   public SqlServerDialect() {
-    super(getSqlTypeMap(), getLogicalToSqlTypeMap(), "[", "]");
+    super("[", "]");
   }
 
   private static Map<Schema.Type, String> getSqlTypeMap() {
```
Comment (willyhoang) [inline src/main/java/io/confluent/connect/jdbc/sink/dialect/SqlServerDialect.java]: I think you forget to delete this map (unless you intentionally left it in for this dialect).

Comment (shikhar) [inline src/main/java/io/confluent/connect/jdbc/sink/dialect/SqlServerDialect.java]: good catch, thanks!

File: src/main/java/io/confluent/connect/jdbc/sink/dialect/DbDialect.java (hunk 1 above)
Comment (shikhar) [inline src/main/java/io/confluent/connect/jdbc/sink/dialect/DbDialect.java]: indeed

**Pull request merged**